stdlib/public/Concurrency/Task.cpp

```diff
@@ -287,6 +298,31 @@ AsyncTaskAndContext swift::swift_task_create_future_f(
     futureContext->indirectResult = futureFragment->getStoragePtr();
   }
 
+  // Perform additional linking between parent and child task.
+  if (parent) {
+    // FIXME: can we allocate this in the task local allocator?
+    // TODO: Do we need to delete it manually or is the record infra handling this?
+    if (flags.task_isGroupChildTask()) {
+      // this task is a child of a task group
+      fprintf(stderr, "[%s:%d] (%s): create group child task record in parent [%d] to %d [group:%d]\n", __FILE__, __LINE__, __FUNCTION__, parent, task, group);
+      swift_task_addStatusRecord(parent,
+                                 new GroupChildTaskStatusRecord(group, task));
+    } else {
+      // just a normal child task
+      fprintf(stderr, "[%s:%d] (%s): create child task record in parent [%d] to %d %d\n", __FILE__, __LINE__, __FUNCTION__, parent, task);
+      swift_task_addStatusRecord(parent,
+                                 new ChildTaskStatusRecord(task));
+    }
+
+    // if the parent was already cancelled, we carry this flag forward to the child.
+    //
+    // In a task group we would not have allowed the `add` to create a child anymore,
+    // however better safe than sorry and `async let` are not expressed as task groups,
+    // so they may have been spawned in any case still.
+    if (swift_task_isCancelled(parent))
+      swift_task_cancel(task);
```

Comment:
the comment explains it well, but I have found this necessary... if we do child tasks (`async let`) in an existing task to set it immediately as cancelled

Reply:
It seems reasonable.